### ISSUE TITLE
[a11y]: ContainedList [with list title decorators] - number next to list title

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13948]
### Package

@carbon/react

### Browser
FF & Safari & iPhone

### Package version

carbon/react@1.30.0

### Automated testing tool and ruleset
Manual QA

### Assistive technology
JAWS & VO

### Description

Would like to know the significance of the number that is displayed after the text "List title"

### WCAG 2.1 Violation
n/a

### Reproduction/example
Record `Number 38` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

### Steps to reproduce

We understand that this number just indicates the number of LI's present in the list.

Need to refine this with the team

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
> We understand that this number just indicates the number of LI's present in the list.

Yeah we need to find a way to link this information to the header so the proper context is read to non-sighted users.

**Comment by tw15egan** [comment]
@tay1orjones I had thought we were looping over each item and generating that number, but it doesn't look like that's the case. It doesn't seem like this is generated by the `ContainedList` at all and is just put there in the story as a tag. I can update the story, but not sure if we wanted to update the component itself to handle this and add the aria attributes automatically.
